### PR TITLE
Add new exercise http salary converter

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
This PR will add a new exercise named http salary converter.

### CODE REVIEW

File: exercises/practice/http-salary-converter/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+* 500 <= `salary` <= 10000
+* `sourceCurrency` and `targetCurrency` are [ISO currency codes (alphabetic)](https://en.wikipedia.org/wiki/ISO_4217#Active_codes). Test input may contain invalid currency codes.
+* The accuracy of the returned rate will be measured up to `5` decimal points.
+
+## Definition
+
+You have to write your code inside the following function.
+
+```ballerina
+# Convert provided salary to local currency
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/.docs/instructions.md:89]:
```suggestion
# Convert provided salary to local currency.
```

File: exercises/practice/http-salary-converter/.meta/config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      "example": [
+        ".meta/reference/main.bal"
+      ]
+    },
+    "icon": "connect",
+    "blurb": "A company needs to convert salaries in foreign currencies to the corresponding local currency using an external API.",
+    "source": "This is an exercise to introduce users to using Ballerina HTTP service data binding",
+    "source_url": "https://ballerina.io/learn/by-example/http-service-data-binding/"
+  }
+
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/.meta/config.json:24]: need new line in here.

File: exercises/practice/http-salary-converter/.meta/reference/main.bal [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import ballerina/http;
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/.meta/reference/main.bal:1]: need a new line after this.

File: exercises/practice/http-salary-converter/.meta/reference/main.bal [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# The exchange rate API base URL
+configurable string apiUrl = "http://localhost:8080";
+
+type Response record {|
+    string base;
+    map<decimal> rates;
+|};
+
+# Convert provided salary to local currency
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/.meta/reference/main.bal:10]:
```suggestion
# Convert provided salary to local currency.
```

File: exercises/practice/http-salary-converter/.meta/reference/main.bal [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#
+# + salary - Salary in source currency
+# + sourceCurrency - Soruce currency
+# + localCurrency - Employee's local currency
+# + return - Salary in local currency or error
+public function convertSalary(decimal salary, string sourceCurrency, string localCurrency) returns decimal|error {
+
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/.meta/reference/main.bal:17]: do we need this line

File: exercises/practice/http-salary-converter/main.bal [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# The exchange rate API base URL
+configurable string apiUrl = "http://localhost:8080";
+
+# Convert provided salary to local currency
```
Comment (SasinduDilshara) [inline exercises/practice/http-salary-converter/main.bal:4]:
```suggestion
# Convert provided salary to local currency.
```

File: exercises/practice/http-salary-converter/.meta/reference/main.bal (hunk 5 above)
Comment (LakshanWeerasinghe) [reply exercises/practice/http-salary-converter/.meta/reference/main.bal:18]: no

**Review by malinthar — APPROVED** [review]

**Pull request merged**